src/zeta/aca_zeta_programming.cpp

```diff
@@ -469,6 +469,25 @@ bool ACA_Zeta_Programming::group_rule_exists(uint group_id)
   }
 }
 
+// Determine whether the gws_ip and gws_mac is right in group entry?
+bool ACA_Zeta_Programming::group_rule_info_correct(uint group_id, string gws_ip, string gws_mac)
```

Comment:
since this function is consumed by test code only, please move it to the test code: test/gtest/aca_test_zeta_programming.cpp